src/ucs/memory/rcache_int.h

```diff
@@ -7,6 +7,8 @@
 #ifndef UCS_REG_CACHE_INT_H_
 #define UCS_REG_CACHE_INT_H_
 
+#include <ucs/type/spinlock.h>
+
```

Comment:
For avoiding this error. I added this part. It may require `extern` part.
Could you tell me where should I write `#include` part?

```
make[4]: Entering directory `/path/to/ucx/test/gtest'
g++ -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/arch/ucx/src -I/home/arch/ucx -I/home/arch/ucx/src -I../../src -I../../test -I../../src -I../../test/gtest  -fopenmp   -I/usr/include -O3 -g -Wall -Werror  -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/arch/ucx/test/gtest/ucm/test_dlopen/.libs\""  -MT ucs/gtest-test_rcache.o -MD -MP -MF ucs/.deps/gtest-test_rcache.Tpo -c -o ucs/gtest-test_rcache.o `test -f 'ucs/test_rcache.cc' || echo './'`ucs/test_rcache.cc
In file included from ucs/test_rcache.cc:13:0:
/home/arch/ucx/src/ucs/memory/rcache_int.h:33:5: error: ‘ucs_spinlock_t’ does not name a type
     ucs_spinlock_t         inv_lock; /**< Lock for inv_q and inv_mp. This is a
     ^
```

Reply:
`src/ucs/Makefile.am` should be updated to include `spinlock.h` into https://github.com/openucx/ucx/blob/c303724fd7bc85ef9609f92c1b5cd575b6a1e1f4/src/ucs/Makefile.am#L18

Reply:
the comment above doesn't make sense anymore
looks like I faced with this error because no `pthread_spinlock_t` on macOS

Reply:
No, I'm building this on CentOS7.

I change Makefile.am and revert `src/ucs/memory/rcache_int.h` and execute the followings
I sill got error message without `#include <ucs/type/spinlock.h>` part.
I'm still not sure why.

```
./autogen.sh
./contrib/configure-devel --disable-numa
make -C test/gtest test V=1
```

```diff
diff --git a/src/ucs/Makefile.am b/src/ucs/Makefile.am
index 98e5dbf..f4f34df 100644
--- a/src/ucs/Makefile.am
+++ b/src/ucs/Makefile.am
@@ -43,6 +43,7 @@ nobase_dist_libucs_la_HEADERS = \
        sys/string.h \
        sys/sock.h \
        sys/stubs.h \
+        type/spinlock.h \
        time/time_def.h \
        type/class.h \
        type/init_once.h \
```


```
g++ -DHAVE_CONFIG_H -I. -I../..  -DCPU_FLAGS="" -I/home/user/ucx/src -I/home/user/ucx -I/home/user/ucx/src -I../../src -I../../test -I../../src -I../../test/gtest  -fopenmp   -I/usr/include -O3 -g -Wall -Werror  -fno-tree-vectorize -DGTEST_UCM_HOOK_LIB_DIR="\"/home/user/ucx/test/gtest/ucm/test_dlopen/.libs\""  -MT ucs/gtest-test_rcache.o -MD -MP -MF ucs/.deps/gtest-test_rcache.Tpo -c -o ucs/gtest-test_rcache.o `test -f 'ucs/test_rcache.cc' || echo './'`ucs/test_rcache.cc
In file included from ucs/test_rcache.cc:13:0:
/home/user/ucx/src/ucs/memory/rcache_int.h:33:5: error: ‘ucs_spinlock_t’ does not name a type
     ucs_spinlock_t         inv_lock; /**< Lock for inv_q and inv_mp. This is a
     ^
```